integration-cli/docker_cli_build_test.go

```diff
@@ -4870,6 +4870,39 @@ func (s *DockerSuite) TestBuildLabelsCache(c *check.C) {
 
 }
 
+func (s *DockerSuite) TestBuildNotVerbose(c *check.C) {
+	// This test makes sure that the `-s` mode works correctly
+	testRequires(c, DaemonIsLinux)
+	tt := []struct {
+		TestName     string // Silent Build Name
+		BuildCmds    string // Build string
+		ShouldFail   bool   // Expect to fail
+		StdoutRegexp string
+		StderrRegexp string
+	}{
+		{"testsilentbuild_busybox_and_echo", "FROM busybox\nRUN echo hello", false, "^(.+:)?[a-z0-9]{64}\\n$", "^$"},
+		{"testsilentbuild_invalid_image", "FROM blabla", true, "^$", "Step 1 : FROM blabla((\\n|\r)+.*)+Error: image library/blabla(:latest)? not found\\n"},
+		{"testsilentbuild_exit_code_1", "FROM busybox\nRUN echo hello\nRUN exit 1", true, "^$", "The command '/bin/sh -c exit 1' returned a non-zero code: 1((\\n|\r)+.*)+Step 1 : FROM busybox((\\n|\r)+.*)+RUN echo hello((\\n|\r)+.*)+hello((\\n|\r)+.*)+Step 3 : RUN exit 1"},
+	}
+
+	for _, te := range tt {
+		_, stdout, stderr, err := buildImageWithStdoutStderr(te.TestName, te.BuildCmds, true, "-q", "--no-cache", "--force-rm", "--rm")
+		if te.ShouldFail && err == nil {
+			c.Fatal(fmt.Errorf("Test [%s] expected to fail but didn't", te.TestName))
+		} else if !te.ShouldFail && err != nil {
+			c.Fatal(fmt.Errorf("Test [%s] shouldn't fail, but got the following error: %s", te.TestName, err.Error()))
+		}
+		outRegexp := regexp.MustCompile(te.StdoutRegexp)
+		if outRegexp.Find([]byte(stdout)) == nil {
+			c.Fatalf("Test [%s] expected stdout to match the [%#v] regexp, but it is [%#v]", te.TestName, te.StdoutRegexp, stdout)
+		}
+		errRegexp := regexp.MustCompile(te.StderrRegexp)
+		if errRegexp.Find([]byte(stderr)) == nil {
+			c.Fatalf("Test [%s] expected stderr to match the [%#v] regexp, but it is [%#v]", te.TestName, te.StderrRegexp, stderr)
+		}
```

Comment:
I like that you check to make sure stderr is empty upon success and stdout is empty upon error!